lib/livebook_web/utils.ex

```diff
@@ -0,0 +1,34 @@
+defmodule LivebookWeb.Utils do
+  @moduledoc "Utility functions for web-focused tasks"
+
+  # Converts human-readable strings to strings which can be used
+  # as HTML element IDs (compatible with HTML4+)
+  #
+  # At the same time duplicate IDs are enumerated to avoid duplicates
+  @spec names_to_ids([binary], MapSet.t()) :: list
+  def names_to_ids(names, used_ids \\ MapSet.new())
+
+  def names_to_ids([], _), do: []
+
+  def names_to_ids([name | rest], used_ids) do
+    id = name_to_id(name)
+
+    id =
+      Stream.iterate(1, &(&1 + 1))
+      |> Stream.map(fn
+        1 -> id
+        index -> "#{id}-#{index}"
+      end)
+      |> Enum.find(&(!MapSet.member?(used_ids, &1)))
+
+    [id | names_to_ids(rest, MapSet.put(used_ids, id))]
+  end
+
+  def name_to_id(name) do
+    name
+    |> String.downcase()
+    |> String.replace(~r/[^\w\-\:\.]+/u, " ")
```

Comment:
Let's ignore `:` and `.` too, they are not particularly meaningful for the anchor href anyway.

Reply:
I think we can keep everything but spaces according to the HTML 5 spec which is safe for us to assumez

Reply:
Nice, makes things much simpler 👍